src/components/ShoppingList/ShoppingList.js

```diff
@@ -41,22 +48,48 @@ function ShoppingList({ listId }) {
     } else {
       return (
         <ul className="shopping-list__list list-reset">
-          {listItems.docs.map((doc) => (
-            <ShoppingListItem
-              key={doc.id}
-              listId={listId}
-              itemId={doc.id}
-              item={doc.data()}
-              handleCheck={handleCheck}
-            />
-          ))}
+          {listItems.docs
+            .filter((doc) => {
+              return doc.data().itemName.includes(filter.toLowerCase());
+            })
+            .map((doc) => (
+              <ShoppingListItem
+                key={doc.id}
+                listId={listId}
+                itemId={doc.id}
+                item={doc.data()}
+                handleCheck={handleCheck}
+              />
+            ))}
         </ul>
       );
     }
   };
 
   return (
     <div className="shopping-list">
+      <div className="filter">
+        <label htmlFor="filterInput" className="filter__label">
+          Filter items
+        </label>
+        <input
+          type="text"
+          id="filterInput"
+          name="filterInput"
+          value={filter}
+          onChange={handleInput}
+          className="filter__text-field text-field"
+        />
+        <button
+          type="button"
+          aria-label="clear"
+          className="filter__button"
+          onClick={() => setFilter('')}
+        >
+          Clear Filter
+        </button>
+      </div>
```

Comment:
Since this isn't needed when there are no list items yet, what do you think of moving this to the `createListElement` function, just above the item list? That way the UI will be more clean and straightforward for a new user.

Reply:
Sure, I just tried it and I'm definitely in favor of having as little information as necessary on the screen at once for a new user